provider/cluster_rosa_classic_resource.go

```diff
@@ -773,10 +845,46 @@ func (r *ClusterRosaClassicResource) Delete(ctx context.Context, request tfsdk.D
 		return
 	}
 
+	// Wait till the cluster has been effectively deleted:
+	if !state.WaitWithTimeout.Unknown && !state.WaitWithTimeout.Null {
+		err = r.waitTillClusterIsNotFound(ctx, state.WaitWithTimeout.Value, resource, r.logger)
+		if err != nil {
+			response.Diagnostics.AddError(
+				"Can't poll cluster state",
+				fmt.Sprintf(
+					"Can't poll state of cluster with identifier '%s': %v",
+					state.ID.Value, err,
+				),
+			)
+			return
+		}
+	}
+
 	// Remove the state:
 	response.State.RemoveResource(ctx)
 }
 
+func (r *ClusterRosaClassicResource) waitTillClusterIsNotFound(ctx context.Context, WaitWithTimeout int64, resource *cmv1.ClusterClient, logger logging.Logger) error {
```

Comment:
Can this and `waitTillClusterIsReady` merged to one function?
Need to add unit tests

Reply:
As we spoke - not really, those two functions use different ocm polling methods- the "create" is using a predicate and the "delete" is using status which checks the status code that returns from the "get" API